portal/models/assessment_status.py

```diff
@@ -186,6 +190,37 @@ def overall_status(self):
                 result = 'In Progress'
         return result
 
+    def withdrawn(self):
+        """Determine if user has `withdrawn` from study """
+        # No easy tie between QB and organization.  Make assumption
+        # that if the QBD isn't intervention tied, the user is associated
+        # with the intervention by org.
+        intervention_qbs = QuestionnaireBank.query.filter(
+            QuestionnaireBank.intervention_id.isnot(None))
+        if self.qbd.questionnaire_bank in intervention_qbs:
+            return False
+
+        # With multiple root organizations, the consent lookup would
+        # be indeterminate - don't allow
+        root_orgs = OrgTree().find_top_level_org(self.user.organizations)
```

Comment:
Out of scope for PR, but these frequently-used queries might benefit from being baked (caching the python -> SQL step). 

[OpenStack - Baked Queries](https://wiki.openstack.org/wiki/OpenStack_and_SQLAlchemy#Baked_Queries)
[SQLAlchemy - Baked Queries](http://docs.sqlalchemy.org/en/latest/orm/extensions/baked.html)